gms/application_state.cc

```diff
@@ -8,48 +8,57 @@
  */
 
 #include "gms/application_state.hh"
-#include <seastar/core/sstring.hh>
-#include <ostream>
-#include <map>
 #include <fmt/ostream.h>
-#include "seastarx.hh"
+#include <seastar/core/sstring.hh>
 
 namespace gms {
 
-static const std::map<application_state, sstring> application_state_names = {
-    {application_state::STATUS,                 "STATUS"},
-    {application_state::LOAD,                   "LOAD"},
-    {application_state::SCHEMA,                 "SCHEMA"},
-    {application_state::DC,                     "DC"},
-    {application_state::RACK,                   "RACK"},
-    {application_state::RELEASE_VERSION,        "RELEASE_VERSION"},
-    {application_state::REMOVAL_COORDINATOR,    "REMOVAL_COORDINATOR"},
-    {application_state::INTERNAL_IP,            "INTERNAL_IP"},
-    {application_state::RPC_ADDRESS,            "RPC_ADDRESS"},
-    {application_state::SEVERITY,               "SEVERITY"},
-    {application_state::NET_VERSION,            "NET_VERSION"},
-    {application_state::HOST_ID,                "HOST_ID"},
-    {application_state::TOKENS,                 "TOKENS"},
-    {application_state::SUPPORTED_FEATURES,     "SUPPORTED_FEATURES"},
-    {application_state::CACHE_HITRATES,         "CACHE_HITRATES"},
-    {application_state::SCHEMA_TABLES_VERSION,  "SCHEMA_TABLES_VERSION"},
-    {application_state::RPC_READY,              "RPC_READY"},
-    {application_state::VIEW_BACKLOG,           "VIEW_BACKLOG"},
-    {application_state::SHARD_COUNT,            "SHARD_COUNT"},
-    {application_state::IGNORE_MSB_BITS,        "IGNOR_MSB_BITS"},
-    {application_state::CDC_GENERATION_ID,      "CDC_STREAMS_TIMESTAMP"}, /* not named "CDC_GENERATION_ID" for backward compatibility */
-    {application_state::SNITCH_NAME,            "SNITCH_NAME"},
-};
+#define APPLICATION_STATE_CASE(state)                                                                                                                          \
```

Comment:
Oh no, preprocessor macros are generally shunned upon in this repository.

Reply:
Another rule that's not mentioned in the coding [style?](https://github.com/scylladb/seastar/blob/master/coding-style.md)

Reply:
Removed the macros. It was merely to avoid repeating the state value twice (and also to prevent possible errors like typos in the name strings). Altogether the Github Copilot generated these things nicely though so the macro is not really needed :)